app/src/main/kotlin/com/wire/android/ui/home/gallery/MediaGalleryViewModel.kt

```diff
@@ -54,9 +69,54 @@ class MediaGalleryViewModel @Inject constructor(
         mediaGalleryViewState = mediaGalleryViewState.copy(screenTitle = conversationName)
     }
 
+    fun saveImageToExternalStorage() {
+        viewModelScope.launch {
+            withContext(dispatchers.io()) {
+                val imageData = getImageData(imageAssetId.conversationId, imageAssetId.messageId)
+                if (imageData is Success) {
+                    val defaultImageName = "Wire downloaded image ${getCurrentParsedDateTime()}.jpeg"
+                    fileManager.saveToExternalStorage(defaultImageName, imageData.decodedAsset) {
+                        onFileSavedToExternalStorage()
+                    }
+                } else {
+                    onSaveError()
+                }
+            }
+        }
+    }
+
+    private fun onSnackbarMessage(messageCode: MediaGallerySnackbarMessages) {
+        viewModelScope.launch {
+            // We need to reset the onSnackbarMessage state so that it doesn't show up again when going -> background -> resume back
+            // The delay added, is to ensure the snackbar message will have enough time to be shown before it is reset to null
+            mediaGalleryViewState = mediaGalleryViewState.copy(onSnackbarMessage = messageCode)
+            delay(ConversationViewModel.SNACKBAR_MESSAGE_DELAY)
+            mediaGalleryViewState = mediaGalleryViewState.copy(onSnackbarMessage = null)
+        }
+    }
+
+    private fun onFileSavedToExternalStorage() {
+        onSnackbarMessage(MediaGallerySnackbarMessages.OnImageDownloaded())
+    }
+
+    private fun onSaveError() {
+        onSnackbarMessage(MediaGallerySnackbarMessages.OnImageDownloadError)
+    }
+
     fun navigateBack() {
         viewModelScope.launch {
             navigationManager.navigateBack()
         }
     }
+
+    fun deleteCurrentImageMessage() {
+        viewModelScope.launch {
+            navigationManager.navigateBack(
+                mapOf(
+                    EXTRA_MESSAGE_TO_DELETE_ID to imageAssetId.messageId,
+                    EXTRA_MESSAGE_TO_DELETE_IS_SELF to imageAssetId.isSelfAsset
+                )
+            )
+        }
+    }
```

Comment:
It would be nice to see tests for this function also. So we have more coverage on the main flows 🙌

Reply:
True that. I just added the main skeleton of the `MediaGalleryViewModelTest` and forgot about adding the other tests 🙈